pkg/distro/defs/id.go

```diff
@@ -7,25 +7,59 @@ import (
 	"github.com/osbuild/images/pkg/distro"
 )
 
+// matchAndNormalize() matches and normalizes the given nameVer
+// based on the reStr. On match it returns the normalized version
+// of the given nameVer.
+func matchAndNormalize(reStr, nameVer string) (string, error) {
+	if reStr == "" {
+		return "", nil
+	}
+
+	re, err := regexp.Compile(`^` + reStr + `$`)
+	if err != nil {
+		return "", fmt.Errorf("cannot use %q: %w", reStr, err)
+	}
+	l := re.FindStringSubmatch(nameVer)
+	switch len(l) {
+	case 0:
+		// no match
+		return "", nil
+	case 1:
+		// simple match, no named matching
+		return nameVer, nil
+	case 2:
+		// incomplete match, user did not provide <name>,<major>,<minor>
+		return "", fmt.Errorf("invalid number of submatches for %q %q (%v)", reStr, nameVer, len(l))
```

Comment:
This could be dropped, right?  Since it's identical to the `default` case.
I suppose we can keep it so that the `2` case is explicitly covered in the list, since dropping it might make it look like we missed a case, but that could be covered by a comment.